src/pcapi/core/offers/api.py

```diff
@@ -282,10 +333,92 @@ def as_utc_without_timezone(d: datetime.datetime) -> datetime.datetime:
+                if offerId == None:
+                    offerId = stock.offerId
+
+                offerer_id = stock.offer.venue.managingOffererId
+                ensure_current_user_has_rights(RightsType.editor, offerer_id)
+                edited_stock = _edit_stock(
+                    stock,
+                    price=data.price,
+                    quantity=data.quantity,
+                    beginning=data.beginning_datetime,
+                    booking_limit_datetime=data.booking_limit_datetime,
+                )
+                edited_stocks.append(edited_stock)
+                stocks.append(edited_stock)
+            else:
+                offerer = offerer_queries.get_by_offer_id(data.offer_id)
+                ensure_current_user_has_rights(RightsType.editor, offerer.id)
+                offer = offer_queries.get_offer_by_id(data.offer_id)
+
+                if offerId == None:
+                    offerId = offer.id
+
+                created_stock = _create_stock(
+                    offer=offer,
+                    price=data.price,
+                    quantity=data.quantity,
+                    beginning=data.beginning_datetime,
+                    booking_limit_datetime=data.booking_limit_datetime,
+                )
+                stocks.append(created_stock)
+        except ApiErrors as e:
+            has_errors = True
+            stock_errors[idx] = e.errors
+
+    if has_errors:
+        db.session.rollback()
+        raise ApiErrors(stock_errors)
+
+    repository.save(*stocks)
+
+    for stock in edited_stocks:
+        previous_beginning = edited_stocks_previous_beginnings[stock.id]
+        if stock.beginningDatetime != previous_beginning:
+            bookings = bookings_repository.find_not_cancelled_bookings_by_stock(stock)
+            if bookings:
+                bookings = update_confirmation_dates(bookings, stock.beginningDatetime)
+                date_in_two_days = datetime.datetime.utcnow() + datetime.timedelta(days=2)
+                check_event_is_in_more_than_48_hours = stock.beginningDatetime > date_in_two_days
+                if check_event_is_in_more_than_48_hours:
+                    bookings = _invalidate_bookings(bookings)
+                try:
+                    user_emails.send_batch_stock_postponement_emails_to_users(
+                        bookings, send_email=mailing.send_raw_email
+                    )
+                except mailing.MailServiceException as exc:
+                    # fmt: off
+                    app.logger.exception(
+                        "Could not notify beneficiaries about update of stock=%s: %s",
+                        stock.id,
+                        exc,
+                    )
+                    # fmt: on
+
+    if feature_queries.is_active(FeatureToggle.SYNCHRONIZE_ALGOLIA):
+        redis.add_offer_id(client=app.redis_client, offer_id=offerId)
+
+    return [*stocks]
```

Comment:
```suggestion
    return stocks
```

ou bien je loupe un truc ?

Reply:
ha oui, j'ai fait de changement, avant j'avais `[*edited_stocks, *created_stocks]`
je vire.